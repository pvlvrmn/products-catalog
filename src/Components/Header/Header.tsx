import {Text} from '@gravity-ui/uikit';
import {TextInput} from "@gravity-ui/uikit";

function Header() {
  return (
    <header className='header__wrapper'>
      <div className='header__title'>
        <Text variant="display-1" color="inherit" ellipsis>Products Shop</Text>
      </div>
      <div className='header__search'>
        <TextInput size='xl' placeholder='Search' hasClear={true}/>
      </div>
    </header>
  )
}

export default Header;
