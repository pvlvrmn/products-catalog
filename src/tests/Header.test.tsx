import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
//import renderer from 'react-test-renderer';
import Header from '../Components/Header/Header.tsx';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

const mockedDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux') as any,
  useDispatch: () => mockedDispatch,
}));

describe('Header', () => {

  test('Header logo render', () => {
    render(<Header />);
    const linkElem = screen.getByText('Products Shop');
    expect(linkElem).toBeInTheDocument();
  });

  it('Header renders correctly', () => {
    const tree = render(<Header />)
    expect(tree).toMatchSnapshot();
  });
})

