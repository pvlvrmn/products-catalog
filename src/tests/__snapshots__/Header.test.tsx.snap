// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header__wrapper"
      >
        <div
          class="header__bars"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1.25 3.25A.75.75 0 0 1 2 2.5h12A.75.75 0 0 1 14 4H2a.75.75 0 0 1-.75-.75Zm0 4.75A.75.75 0 0 1 2 7.25h12a.75.75 0 0 1 0 1.5H2A.75.75 0 0 1 1.25 8ZM2 12a.75.75 0 0 0 0 1.5h12a.75.75 0 0 0 0-1.5H2Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="header__title"
        >
          <span
            class="yc-text yc-text_variant_display-1 yc-text_ellipsis"
          >
            Products Shop
          </span>
        </div>
        <div
          class="header__search"
        >
          <span
            class="yc-text-input yc-text-input_view_normal yc-text-input_size_xl yc-text-input_pin_round-round"
          >
            <span
              class="yc-text-input__content"
            >
              <input
                class="yc-text-input__control yc-text-input__control_type_input"
                placeholder="Search"
                value=""
              />
            </span>
          </span>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header__wrapper"
    >
      <div
        class="header__bars"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1.25 3.25A.75.75 0 0 1 2 2.5h12A.75.75 0 0 1 14 4H2a.75.75 0 0 1-.75-.75Zm0 4.75A.75.75 0 0 1 2 7.25h12a.75.75 0 0 1 0 1.5H2A.75.75 0 0 1 1.25 8ZM2 12a.75.75 0 0 0 0 1.5h12a.75.75 0 0 0 0-1.5H2Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="header__title"
      >
        <span
          class="yc-text yc-text_variant_display-1 yc-text_ellipsis"
        >
          Products Shop
        </span>
      </div>
      <div
        class="header__search"
      >
        <span
          class="yc-text-input yc-text-input_view_normal yc-text-input_size_xl yc-text-input_pin_round-round"
        >
          <span
            class="yc-text-input__content"
          >
            <input
              class="yc-text-input__control yc-text-input__control_type_input"
              placeholder="Search"
              value=""
            />
          </span>
        </span>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
